### ISSUE TITLE
[BUG] Calling 'fresh' under heavy load causes the upstream to close unexpectedly

### ISSUE THREAD

**Issue by lukisk** [MobileNativeFoundation/Store#636]
**Describe the bug**
Occasionally, the 'main stream' closes unexpectedly.

**To Reproduce**
```
val scope = CoroutineScope(Dispatchers.Default)

val store = StoreBuilder.from<Int, String>(Fetcher.of { key ->
    delay(10) //simulate long network request
    "$key :: " + Random.nextInt()
}).build()

scope.launch {
    store.stream(StoreReadRequest.cached(1, false)) //main stream
        .onCompletion {
            println("Unexpected !!!" + it)
        }
        .collect {
            println("result" + it)
    }
}

(1 .. 50).onEach  {
    scope.launch {
        while (isActive) {
            store.stream(StoreReadRequest.fresh(1))
                .filterNot { it is StoreReadResponse.Loading }
                .firstOrNull()
            delay(150)
        }
    }
}
```
- after while, the 'main stream' is completed unexpectedly
-  behavior is same when Store.fresh is called

**Expected behavior**
Calling 'fresh' does not cause the 'main stream' to close

**Smartphone (please complete the following information):**
 - Device: all
 - OS: all
 - Store Version 4.*, 5.*

**Comment by matt-ramotar** [comment]
Hey @andrewfarm, thanks for following up. Let me get fixes in for #722 and #660 and then we can start beta -> stable

**Comment by matt-ramotar** [comment]
I merged #727 and #728 to fix #722 and #660. Will release 5.1.0-alpha08 (#729) and then start beta -> stable next week
